test/models/course_queue_test.rb

```diff
@@ -49,6 +50,197 @@ class CourseQueueTest < ActiveSupport::TestCase
     assert @queue.outstanding_requests[-1] == last_entry
   end
 
+  test "request are sorted by number of previously resolved requests" do
+
+    assert @queue_sort.course_queue_entries.blank?
+    assert @queue_sort.course.sort_by_number_of_resolved_requests_this_session?
+
+    entry_sue_1 = @queue_sort.request(
+      requester: users(:sue),
+      description: 'sue 1',
+      location: '',
+      group: nil
+    )
+
+    entry_sue_1.resolve_by!(users(:matt))
+
+    assert @queue_sort.outstanding_requests.where(resolved_at: nil).blank?
+
+    entry_sue_2 = @queue_sort.request(
+      requester: users(:sue),
+      description: 'sue 2',
+      location: '',
+      group: nil
+    )
+
+    entry_sue_2.resolve_by!(users(:matt))
+
+    assert @queue_sort.outstanding_requests.where(resolved_at: nil).blank?
+
+    entry_sue_3 = @queue_sort.request(
+      requester: users(:sue),
+      description: 'sue 3',
+      location: '',
+      group: nil
+    )
+
+    entry_sue_3.resolve_by!(users(:matt))
+
+    assert @queue_sort.outstanding_requests.where(resolved_at: nil).blank?
+
+    entry_sue_4 = @queue_sort.request(
+      requester: users(:sue),
+      description: 'sue 4',
+      location: '',
+      group: nil
+    )
+
+    entry_jim_1 = @queue_sort.request(
+      requester: users(:jim),
+      description: 'jim 1',
+      location: '',
+      group: nil
+    )
+
+    entry_jim_1.resolve_by!(users(:matt))
+
+    entry_steve_1 = @queue_sort.request(
+      requester: users(:steve),
+      description: 'steve 1',
+      location: '',
+      group: nil
+    )
+
+    assert @queue_sort.outstanding_requests[0] == entry_steve_1
+
+    entry_steve_1.resolve_by!(users(:matt))
+
+    entry_steve_2 = @queue_sort.request(
+      requester: users(:steve),
+      description: 'steve 2',
+      location: '',
+      group: nil
+    )
+
+    entry_jim_2 = @queue_sort.request(
+      requester: users(:jim),
+      description: 'jim 2',
+      location: '',
+      group: nil
+    )
+
+    entry_jim_2.resolve_by!(users(:matt))
+
+    entry_jim_3 = @queue_sort.request(
+      requester: users(:jim),
+      description: 'jim 3',
+      location: '',
+      group: nil
+    )
+
+    assert @queue_sort.outstanding_requests[0] == entry_steve_2
+    assert @queue_sort.outstanding_requests[1] == entry_jim_3
+    assert @queue_sort.outstanding_requests[2] == entry_sue_4
+
+    # On tie, sort by timestamp
+
+    entry_jim_3.resolve_by!(users(:matt))
+
+    entry_jim_4 = @queue_sort.request(
+      requester: users(:jim),
+      description: 'jim 4',
+      location: '',
+      group: nil
+    )
+
+    assert @queue_sort.outstanding_requests[0] == entry_steve_2
+    assert @queue_sort.outstanding_requests[1] == entry_sue_4
+    assert @queue_sort.outstanding_requests[2] == entry_jim_4
+
+    # close queue and reopen, should "reset"
+    # need to sleep to make timestamps different
+    sleep(3)
```

Comment:
Tests that depend on time are notoriously brittle, and sleeping during tests increases the runtime of the suite. I know this test suite isn't the best, but I'd like to avoid adding `sleep`s. Perhaps we could use some of the [Rails time travel helpers](https://api.rubyonrails.org/v5.1.7/classes/ActiveSupport/Testing/TimeHelpers.html)?